### PR TITLE
feat(ts): Typescript examples + Improve `vue-app` typings

### PR DESCRIPTION
**Pull request by kevinmarrec**
## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description
This PR follows https://github.com/nuxt/nuxt.js/pull/4658

It provides : 
- 2 Examples : `typescript` & `typescript-vuex`
- `vue-app` Typings update : `process.server`, `process.client`, `process.browser`, `process.static`

With current TS state, only documentation should be needed to have the feature ready for release.

Here is some things that can be done but not mandatory (IMO) for feature release :
- Typings for Nuxt configuration file, modules, plugins and everything else that should be typed in the Nuxt application
- Test improvements regarding TypeScript fixtures

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [ ] All new and existing tests are passing.

### CODE REVIEW

**Comment by manniL** [comment]
@kevinmarrec seems like old/duplicate commits from #4658 are included :thinking:

**Comment by kevinmarrec** [comment]
@manniL I think it's normal as it's the same branche ? I mean commits are referenced but already merged, aren't they ?

**Comment by manniL** [comment]
image

Shouldn't be like that *I think*.
maybe because the commits before are squashed into ee0096bf594d2ef17b611f4975e6ce3a8a199b76 now.

**Comment by kevinmarrec** [comment]
Well I don't know what to do then 🙈

**Comment by manniL** [comment]
@kevinmarrec I guess it doesn't matter too much as the content of the old commits is already in anway. Just harms the overview a little but :man_shrugging: :relaxed:

**Comment by kevinmarrec** [comment]
Alright 😁

File: examples/typescript-vuex/nuxt.config.ts [hunk 1]
```diff
@@ -1,23 +1,15 @@
...
-    baseUrl: process.env.BASE_URL || 'http://localhost:3000'
-  },
+const config = {
   head: {
     title: 'starter',
     meta: [
       { charset: 'utf-8' },
       { name: 'viewport', content: 'width=device-width, initial-scale=1' },
-      { hid: 'description', name: 'description', content: 'Nuxt.js project' }
+      { hid: 'description', name: 'description', content: 'Nuxt TS project' }
```
Comment (pi0) [inline examples/typescript-vuex/nuxt.config.ts]: Nuxt.ts :))

**Review by pi0 — APPROVED** [review]

**Pull request merged**